xarray/util/deprecation_helpers.py

```diff
@@ -115,3 +124,26 @@ def inner(*args, **kwargs):
         return inner
 
     return _decorator
+
+
+def deprecate_dims(func: Callable[P, R]) -> Callable[P, R]:
```

Comment:
You could also use the TypeVar named `T` defined above instead of importing ParamSpec here.
```suggestion
def deprecate_dims(func: T) -> T:

```

Reply:
Ofc, thank you!